dispatcher/examples/gemm/cpp/09_multi_registry.cpp

```diff
@@ -0,0 +1,208 @@
+            config.build_key(), KERNEL_NAME);
+
+    compute_registry.register_kernel(kernel);
+    memory_registry.register_kernel(kernel);
+    latency_registry.register_kernel(kernel);
+
+    std::cout << "  " << compute_registry.get_name() << ": " << compute_registry.size()
+              << " kernel(s)\n";
+    std::cout << "  " << memory_registry.get_name() << ": " << memory_registry.size()
+              << " kernel(s)\n";
+    std::cout << "  " << latency_registry.get_name() << ": " << latency_registry.size()
+              << " kernel(s)\n";
+
+    // =========================================================================
+    // Create dispatchers for each registry
+    // =========================================================================
+    Dispatcher compute_dispatcher(&compute_registry);
+    Dispatcher memory_dispatcher(&memory_registry);
+    Dispatcher latency_dispatcher(&latency_registry);
+
+    // =========================================================================
+    // Run with different dispatchers
+    // =========================================================================
+    std::cout << "\nRunning with different dispatchers:\n";
+    print_separator();
+
+    struct WorkloadTest
+    {
+        const char* name;
+        Dispatcher* dispatcher;
+        int M, N, K;
+    };
+
+    std::vector<WorkloadTest> tests = {
+        {"Compute-bound", &compute_dispatcher, 4096, 4096, 4096},
+        {"Memory-bound", &memory_dispatcher, 1024, 1024, 1024},
+        {"Latency-opt", &latency_dispatcher, 512, 512, 512},
+    };
+
+    bool all_passed = true;
+
+    for(const auto& test : tests)
+    {
+        Problem problem(test.M, test.N, test.K);
+
+        GpuBuffer<ADataType> a_dev(test.M * test.K);
+        GpuBuffer<BDataType> b_dev(test.K * test.N);
+        GpuBuffer<CDataType> c_dev(test.M * test.N);
+
+        std::vector<ADataType> a_host(test.M * test.K, ADataType(1.0f));
+        std::vector<BDataType> b_host(test.K * test.N, BDataType(1.0f));
+        a_dev.copy_from_host(a_host.data());
+        b_dev.copy_from_host(b_host.data());
+        c_dev.zero();
+
+        float time_ms =
+            test.dispatcher->run(a_dev.get(), b_dev.get(), c_dev.get(), problem, nullptr);
+        double tflops = calculate_tflops(test.M, test.N, test.K, time_ms);
+
+        std::cout << test.name << " (" << test.M << "x" << test.N << "x" << test.K << "):\n";
+        std::cout << "  Time:   " << std::fixed << std::setprecision(4) << time_ms << " ms\n";
+        std::cout << "  TFLOPS: " << std::setprecision(2) << tflops << "\n";
+
+        // Verify
+        std::vector<CDataType> c_host(test.M * test.N);
+        c_dev.copy_to_host(c_host.data());
+        float expected = static_cast<float>(test.K);
+        // Use 1% relative tolerance for FP16 accumulation over K elements
+        if(std::abs(static_cast<float>(c_host[0]) - expected) > (0.01f * expected + 1.0f))
```

Comment:
also for the correctness check it would be nice to set atol/rtol and not hardcode them